csgo/items.go

```diff
@@ -370,6 +389,130 @@ func mapToStickerCapsule(index int, data map[string]interface{}, stickers []stri
 	return response, nil
 }
 
+// Tool represents consumable inventory only items
+type Tool struct {
+	Id    string `json:"id"`
+	Index int    `json:"index"`
+	Name  string `json:"name"`
+}
+
+// mapToTool converts the provided map into a Tool providing
+// all required parameters are present and of the correct type.
+func mapToTool(index int, data map[string]interface{}, language *language) (*Tool, error) {
+	response := &Tool{
+		Index: index,
+	}
+	if val, err := crawlToType[string](data, "name"); err != nil {
+		return nil, errors.Wrap(err, "unable to crawl Tool item to path: name")
+	} else {
+		response.Id = val
+	}
+
+	if val, err := crawlToType[string](data, "item_name"); err == nil {
+		lang, _ := language.lookup(val)
+		if lang == "" {
+			lang, _ = language.lookup(val[1:])
+		}
+		response.Name = lang
+	} else if val, err := crawlToType[string](data, "item_type_name"); err == nil {
+		lang, _ := language.lookup(val)
+		if lang == "" {
+			lang, _ = language.lookup(val[1:])
+		}
+		response.Name = lang
+	}
+
+	if response.Name == "" {
+		return nil, errors.New("unable to locate Tools's language Name Id" + fmt.Sprintf("%+v", response))
+	}
+
+	return response, nil
+}
+
+// Character represents a skin that a player can use ingame
+type Character struct {
+	Id          string `json:"id"`
+	Index       int    `json:"index"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	RarityId    string `json:"rarityId"`
+}
+
+// mapToCharacter converts the provided map into a Character providing
+// all required parameters are present and of the correct type.
+func mapToCharacter(index int, data map[string]interface{}, language *language) (*Character, error) {
+	response := &Character{
+		Index: index,
+	}
+	if val, err := crawlToType[string](data, "name"); err != nil {
+		return nil, errors.Wrap(err, "unable to crawl Tool item to path: name")
+	} else {
+		response.Id = val
+	}
+
+	if val, err := crawlToType[string](data, "item_name"); err == nil {
+		lang, _ := language.lookup(val[1:])
+		response.Name = lang
+	}
+
+	if val, err := crawlToType[string](data, "item_description"); err == nil {
+		lang, _ := language.lookup(val[1:])
+		response.Description = lang
+	}
+
+	if val, err := crawlToType[string](data, "item_rarity"); err == nil {
+		response.RarityId = val
+	}
+
+	if response.Name == "" {
+		return nil, errors.New("unable to locate Character's language Name Id" + fmt.Sprintf("%+v", response))
+	}
+	if response.Description == "" {
+		return nil, errors.New("unable to locate Character's language Description Id" + fmt.Sprintf("%+v", response))
+	}
+	if response.RarityId == "" {
+		return nil, errors.New("unable to locate Character's language Rarity Id" + fmt.Sprintf("%+v", response))
+	}
```

Comment:
Just a suggestion as it's a bit more succinct (there may be one or two other places also, but merely a suggestion - it's functional as-is):
```suggestion
	if response.Name == "" {
		return nil, fmt.Errorf("unable to locate Character's language Name Id: %+v", response)
	}
	if response.Description == "" {
		return nil, fmt.Errorf("unable to locate Character's language Description Id: %+v", response)
	}
	if response.RarityId == "" {
		return nil, fmt.Errorf("unable to locate Character's language Rarity Id: %+v", response)
	}
```